config/routes.rb

```diff
@@ -7,7 +7,11 @@
 
   resources :adoptable_pets, only: [:index, :show]
   resources :adopter_applications, only: [:index, :edit, :update], controller: "adoption_application_reviews"
-  resource :adopter_profile, except: :destroy, as: "profile"
+  resource :adopter_profile, except: :destroy, as: "profile" do
+    collection do
+      get "states"
+    end
```

Comment:
I think you can make a general endpoint outside of the `adopter_profile` to generalize it so it can be used in other spots. Check my other comment about making this more RESTful